src/fallback.rs

```diff
@@ -32,16 +32,25 @@ pub static FISH_EMOJI_WIDTH: AtomicIsize = AtomicIsize::new(1);
 static WC_LOOKUP_TABLE: Lazy<WcLookupTable> = Lazy::new(WcLookupTable::new);
 
 /// A safe wrapper around the system `wcwidth()` function
+#[cfg(not(target_os = "cygwin"))]
 pub fn wcwidth(c: char) -> isize {
     extern "C" {
         pub fn wcwidth(c: libc::wchar_t) -> libc::c_int;
     }
 
-    const _: () = assert!(mem::size_of::<libc::wchar_t>() >= mem::size_of::<char>());
+    const _: () = assert!(std::mem::size_of::<libc::wchar_t>() >= std::mem::size_of::<char>());
     let width = unsafe { wcwidth(c as libc::wchar_t) };
     isize::try_from(width).unwrap()
 }
 
+/// A safe wrapper around the system `wcwidth()` function
+#[cfg(target_os = "cygwin")]
+pub fn wcwidth(c: char) -> isize {
```

Comment:
What you're doing here is adding a dependency on a unicode width table. Fish already ships its own, so there's no need for a second one.

We call wcwidth in two places:

https://github.com/fish-shell/fish-shell/blob/90218821f9a373684b439abe5cf200dc467fa7fe/src/env_dispatch.rs#L200-L204 - which figures out if Unicode >= 9 is supported. That one can just be hardcoded as "2".

https://github.com/fish-shell/fish-shell/blob/90218821f9a373684b439abe5cf200dc467fa7fe/src/fallback.rs#L51-L53, as an attempt to support "the system console", i.e. something like linux' or freebsd's kernel VTs. That does not apply to Cygwin.

So this can be removed entirely.

Reply:
Seems that there's another usage:
https://github.com/fish-shell/fish-shell/blob/90218821f9a373684b439abe5cf200dc467fa7fe/src/fallback.rs#L75-L78
Should I hardcode 0 here?

Reply:
Yeah, that sounds right.